fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/AvroDimensionRowParser.java

```diff
@@ -93,34 +95,45 @@ private boolean doesSchemaContainAllDimensionFields(Dimension dimension, Schema
      *
      * @throws IllegalArgumentException thrown if JSON object `fields` is not present
      */
-    public Set<DimensionRow> parseAvroFileDimensionRows(Dimension dimension, String avroFilePath)
+    public Stream<DimensionRow> parseAvroFileDimensionRows(Dimension dimension, String avroFilePath)
         throws IllegalArgumentException {
 
         // Creates an AVRO DatumReader object
         DatumReader<GenericRecord> datumReader = new GenericDatumReader<>();
 
         // Creates an AVRO DataFileReader object that reads the AVRO data file one record at a time
-        try (DataFileReader<GenericRecord> dataFileReader = new DataFileReader<>(new File(avroFilePath), datumReader)) {
+        try {
+            DataFileReader<GenericRecord> dataFileReader = new DataFileReader<>(new File(avroFilePath), datumReader);
             // Validate Schema
             if (!doesSchemaContainAllDimensionFields(dimension, dataFileReader.getSchema())) {
                 String msg = "The AVRO schema file does not contain all the configured dimension fields";
                 LOG.error(msg);
                 throw new IllegalArgumentException(msg);
             }
-
+            Function<GenericRecord, Map<String, String>> recordMapFunction = genericRecord -> dimension
+                    .getDimensionFields()
+                    .stream()
+                    .collect(
+                            Collectors.toMap(
+                                    DimensionField::getName,
+                                    dimensionField -> resolveRecordValue(
+                                            genericRecord, dimensionFieldNameMapper.convert(dimension, dimensionField))
+                            )
+                    );
+            Runnable fileCloser = () -> {
+                try {
+                    dataFileReader.close();
+                } catch (IOException e) {
+                    String msg = String.format("Error closing avro file, at the location %s", avroFilePath);
```

Comment:
Generally, if the file is already closed when the closer gets called, any exception that would be generated should probably be swallowed and not propagated. Other errors other than file already closed should propagate. This prevents weird errors from occuring if the file is closed multitple times.